--- a/linux-sunxi/arch/arm/mach-sunxi/power/brom/Makefile
+++ b/linux-sunxi/arch/arm/mach-sunxi/power/brom/Makefile
@@ -5,7 +5,7 @@ targets := resumes.elf
 
 #use "-Os" flags.
 #Don't use "-O2" flags.
-KBUILD_CFLAGS := -g -c -nostdlib -march=armv7-a -marm -fno-unwind-tables -fno-asynchronous-unwind-tables -mlittle-endian -O2 --min_array_alignment=4 --no_unaligned_access
+KBUILD_CFLAGS := -g -c -nostdlib -march=armv7-a -marm -fno-unwind-tables -fno-asynchronous-unwind-tables -mlittle-endian -O2
 
 #Include the cur dir.
 KBUILD_CPPFLAGS += -I.
--- a/linux-sunxi/arch/arm/mm/dma-mapping.c
+++ b/linux-sunxi/arch/arm/mm/dma-mapping.c
@@ -576,7 +576,7 @@ static void __free_from_contiguous(struct device *dev, struct page *page,
 	dma_release_from_contiguous(dev, page, size >> PAGE_SHIFT);
 }
 
-inline pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot)
+pgprot_t __get_dma_pgprot(struct dma_attrs *attrs, pgprot_t prot)
 {
 	prot = dma_get_attr(DMA_ATTR_WRITE_COMBINE, attrs) ?
 			    pgprot_writecombine(prot) :
